firmware/src/networking/wifi/esp.rs

```diff
@@ -67,7 +68,7 @@ pub async fn poll_dhcp(net: &mut Network<'_>) {
 	// wait for getting an ip address
 	debug!("Wait to get an ip address");
 	loop {
-		net.poll_dhcp().unwrap();
+		net.poll_dhcp().expect("failed to poll dhcp");
```

Comment:
Maybe this just yield + continue in the loop if it fails instead?

Reply:
I need to investigate what would cause an Err tbh